Path/path.cpp

```diff
@@ -29,24 +41,33 @@ std::vector<Vertex> PatrolPath::GetPointsAsVertices() {
 }
 
 bool PatrolPath::IsCurrentWaypointReached(glm::vec3 position) {
-    glm::vec3 currentWaypointPosition = m_Points[ m_CurrentWaypointIndex ].position;
+    const auto& currentWaypointEntry = m_TargetnameToWaypoint.find( m_CurrentWaypointName );
+    if ( currentWaypointEntry == m_TargetnameToWaypoint.end() ) {
+        return false;
+    }
+    const Waypoint& waypoint = currentWaypointEntry->second;
+    glm::vec3 currentWaypointPosition = waypoint.position;
     float distance = glm::distance(position, currentWaypointPosition);
     //printf("Distance to waypoint: %f\n", distance);
+    
     return distance < m_Radius;
 }
 
 void PatrolPath::TargetNextWaypoint() {
-    m_PreviousWaypointIndex = m_CurrentWaypointIndex;
-    m_CurrentWaypointIndex = m_NextWaypointIndex;
-    m_NextWaypointIndex += 1;
-    if ( m_NextWaypointIndex >= m_Points.size() ) {
-        m_NextWaypointIndex = 0;
-    }
-    // if ( m_NextWaypointIndex < 0 ) {
-    //     m_NextWaypointIndex = m_Points.size() - 1;
-    // }
+    Waypoint currentWaypoint = GetCurrentWaypoint();
+    m_PreviousWaypointName = m_CurrentWaypointName;
+    m_CurrentWaypointName = currentWaypoint.target;
+    printf("Setting waypoint to: %s\n", m_CurrentWaypointName.c_str());
 }
```

Comment:
Good question. I just implemented something where each waypoint would have a `parent` string. But that results in a much more complex setup loop in `CWorld` and so on... At the moment we just don't support it, I would suggest, unless you absolutely need it. As I want to have waypoints as entities themselves later, I'd rather wait until we change that. Then I would probably represent waypoints as a doubly linked list instead of having to use a map.